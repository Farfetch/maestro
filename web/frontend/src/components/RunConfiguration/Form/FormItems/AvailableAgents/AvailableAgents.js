import { Form } from "antd";
import { useEffect, useState } from "react";

import { fetchAgents } from "../../../../../lib/api/endpoints/agent";
import AgentsSelect from "../../../../form/select/AgentsSelect";

const AvailableAgents = () => {
  const [agents, setAgents] = useState([]);

  const updateAgentsList = async () => {
    const result = await fetchAgents();

    setAgents(result);
  };

  useEffect(() => {
    updateAgentsList();
  }, []);

  return (
    <>
      <Form.Item
        label="Client Agent"
        name="clientAgentId"
        rules={[{ required: true, message: "Please select client Agent!" }]}
      >
        <AgentsSelect mode="single" agents={agents} />
      </Form.Item>
      <Form.Item
        label="Server Agents"
        name="serverAgentIds"
        rules={[
          {
            required: true,
            message: "Please select at least one of server agents!"
          }
        ]}
      >
        <AgentsSelect mode="multiple" agents={agents} />
      </Form.Item>
    </>
  );
};

export default AvailableAgents;
