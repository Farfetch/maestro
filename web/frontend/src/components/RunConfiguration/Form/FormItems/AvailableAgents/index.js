import AvailableAgents from "./AvailableAgents";

export default AvailableAgents;
