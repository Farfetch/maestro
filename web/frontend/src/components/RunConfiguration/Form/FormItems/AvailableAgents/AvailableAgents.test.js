import { cleanup, render, waitFor } from "@testing-library/react";
import { Form } from "antd";
import React from "react";

import AvailableAgents from "./AvailableAgents";

jest.mock("../../../../../lib/api/endpoints/agent", () => ({
  fetchAgents: () => [
    {
      id: "agent-id",
      hostname: "agent-hostname",
      ip: "test-ip",
      agentStatus: "AVAILABLE"
    }
  ]
}));

afterEach(cleanup);

describe("components/RunConfiguration/Form/AvailableAgents", () => {
  test(`should render AvailableAgents component`, async () => {
    const { container } = render(
      <Form>
        <AvailableAgents />
      </Form>
    );

    await waitFor(() => {
      expect(container.outerHTML).toMatchSnapshot();
    });
  });
});
