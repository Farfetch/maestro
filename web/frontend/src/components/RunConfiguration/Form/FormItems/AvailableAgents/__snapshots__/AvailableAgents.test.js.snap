// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RunConfiguration/Form/AvailableAgents should render AvailableAgents component 1`] = `"<div><form class=\\"ant-form ant-form-horizontal\\"><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-label\\"><label for=\\"clientAgentId\\" class=\\"ant-form-item-required\\" title=\\"Client Agent\\">Client Agent</label></div><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><div class=\\"ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow\\" style=\\"width: 100%;\\"><div class=\\"ant-select-selector\\"><span class=\\"ant-select-selection-search\\"><input autocomplete=\\"off\\" type=\\"search\\" class=\\"ant-select-selection-search-input\\" style=\\"opacity: 0;\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-owns=\\"rc_select_TEST_OR_SSR_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"rc_select_TEST_OR_SSR_list\\" aria-activedescendant=\\"rc_select_TEST_OR_SSR_list_0\\" readonly=\\"\\" unselectable=\\"on\\" value=\\"\\" id=\\"rc_select_TEST_OR_SSR\\"></span><span class=\\"ant-select-selection-placeholder\\">Select Agent from the list</span></div><span class=\\"ant-select-arrow\\" style=\\"user-select: none;\\" unselectable=\\"on\\" aria-hidden=\\"true\\"><span role=\\"img\\" aria-label=\\"down\\" class=\\"anticon anticon-down ant-select-suffix\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z\\"></path></svg></span></span></div></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-label\\"><label for=\\"serverAgentIds\\" class=\\"ant-form-item-required\\" title=\\"Server Agents\\">Server Agents</label></div><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><div class=\\"ant-select ant-select-multiple ant-select-allow-clear ant-select-show-search\\" style=\\"width: 100%;\\"><div class=\\"ant-select-selector\\"><div class=\\"ant-select-selection-overflow\\"><div class=\\"ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix\\" style=\\"opacity: 1;\\"><div class=\\"ant-select-selection-search\\" style=\\"width: 0px;\\"><input autocomplete=\\"off\\" type=\\"search\\" class=\\"ant-select-selection-search-input\\" style=\\"opacity: 0;\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-owns=\\"rc_select_TEST_OR_SSR_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"rc_select_TEST_OR_SSR_list\\" aria-activedescendant=\\"rc_select_TEST_OR_SSR_list_0\\" readonly=\\"\\" unselectable=\\"on\\" value=\\"\\" id=\\"rc_select_TEST_OR_SSR\\"><span class=\\"ant-select-selection-search-mirror\\" aria-hidden=\\"true\\">&nbsp;</span></div></div></div><span class=\\"ant-select-selection-placeholder\\">Select Agent from the list</span></div></div></div></div></div></div></form></div>"`;
